### DISCUSSION TITLE
Future of tss-react with respect to Pigment

### DISCUSSION THREAD

**Discussion by ptbrowne** [garronej/tss-react#213] [General]
Hi,

I have been a happy user of tss-react, and also was waiting to do CSS extracting at build time. Now that the MUI team has released a first version of their [zero-runtime CSS solution](https://github.com/mui/material-ui/tree/master/packages/pigment-css-react), I was wondering how did you see tss-react evolving ?

- Will it somehow evolve to become compatible ?
- Will we be able to do zero-runtime with tss-react ?

Thanks a lot for tss-react :)

**Comment by garronej** [comment]
Hello @ptbrowne,  
Thanks for being a tss-react user.  

When MUI v6 and v7 releases I'll update the documentation and the `tss-react/mui` export. It will remain compatible, the fact that MUI abandons emotion in favor of a inhouse CSS solution should not be a problem.  

> Will we be able to do zero-runtime with tss-react ?

No, tss-react will never be a zero-runtime solution. There are multiple alternative such as stylex, vanilla-extract and Panda that implement static style extraction if you'd like to go this way.  
I strongly belive than having style generated at runtime enable a level of flexibility, expressiveness and maintainability that any solutions based on statically generated styles will never be able to match and that the performance argument isn't very strong.  

There are two pain point however that I hope we'll eventually be able to address:  
- White flashes with SSR. It's particulary noticable [on this page](https://mui.com/material-ui/material-icons/) (if you have dark mode preference on your OS.) This is due to the fact that we must must do the switch from light to dark in a useEffect callback or we get idratation errors.  
- Lack of support for Server Component. It is not nessearely a lost battle [ref](https://www.joshwcomeau.com/react/css-in-rsc/).  

If there's some breakthough on thoses front I'll implement them. If not tss-react will remain a strong solution for SPAs and will remain compatible with Next and SSR although with the above mentioned limitations.

**Reply by ptbrowne** [reply]
Thanks for the fast reply and the clarification 👍

**Reply by cjnoname** [reply]
Hey mate, I'm another happy user here—thanks for building this great library! 

I was just wondering: what happens if MUI drops Emotion support in v8? Is there a way to make tss-react work with the new "Pigment" CSS?

Thanks!
